components/fxa-client/src/internal/http_client.rs

```diff
@@ -22,58 +28,68 @@ const HAWK_HKDF_SALT: [u8; 32] = [0b0; 32];
 const HAWK_KEY_LENGTH: usize = 32;
 const RETRY_AFTER_DEFAULT_SECONDS: u64 = 10;
 
+/// Trait defining the low-level API for talking to the FxA server.
+///
+/// These are all the methods and datatypes used for interacting with
+/// the FxA server. It's defined as a trait mostly to make it mockable
+/// for testing purposes.
```

Comment:
This trait, on the other hand, I just could not resist trying to clean up a little while I was in here...